### ISSUE TITLE
Rearranged drives, now even newly selected .exes are 

### ISSUE THREAD

**Issue by alphaniner** [nefarius/HidHide#122]
- [ ] I have searched open and closed issues for duplicates
- [ ] I am submitting a bug report for existing functionality that does not work as intended
- [ ] I have read [the project documentation](https://docs.nefarius.at/projects/) and not found a solution to my problem there
- [ ] This isn't a feature request or a discussion topic

----------------------------------------

### Bug description

Describe here the issue that you are experiencing.

### Steps to reproduce

- using hyphens as bullet points
- list the steps
- that reproduce the bug

**Actual result:** Describe here what happens after you run the steps above (i.e. the buggy behaviour)
**Expected result:** Describe here what should happen after you run the steps above (i.e. what would be the correct behaviour)

### Machine info

**CPU-Architecture:** x86, x64, ARM, ARM64, ...
**Windows version:** Windows 10 20H2
**Software/driver version(s):** 1.4.14

**Comment by alphaniner** [comment]
Can you provide any advice on a workaround for this in the interim? I went so far as to delete, recreate, and reformat the partition, and in Device Manager the volume shows **Physical Device Object name** `\Device\HarddiskVolume6`. But when I add an exe on the partition, it _still_ shows up as `\Device\HarddiskVolume10\...` and is pre-selected.

**Comment by nefarius** [comment]
I don't know what the issue is, so no.

**Comment by alphaniner** [comment]
Is there a way to _completely_ purge all configuration data? Seems to me that would be worthwhile to try. IIRC in another bug report you said config is stored in the registry; I was poking around and came across a couple keys I couldn't access. Also IIRC you said there that the config is not user-accessible by design (whether yours or Microsoft's wasn't clear). If those are it, I will try to find a way to access/delete them.

**Comment by nefarius** [comment]
Yes there is indeed; caution though, your own responsibilities lie ahead 😉 

- Get the MS tool `PsExec`
- Open a CMD or Terminal as Admin 
- Run `.\psexec -s -i regedit`
- Now Registry Editor should be open with system permissions
- Now go to `Computer\HKEY_LOCAL_MACHINE\SYSTEM\CurrentControlSet\Services\HidHide\Parameters` and adjust the key permissions like so:  
  ![image](https://github.com/nefarius/HidHide/assets/286631/29edf02c-31b6-463c-bc42-7608db5c14c2)  
  ![image](https://github.com/nefarius/HidHide/assets/286631/43acc85d-27f9-4641-a8c1-afc149d23eda)
- Now you should be able to see those:  
  ![image](https://github.com/nefarius/HidHide/assets/286631/7fcf0b01-93af-4702-b3ef-335b2a1fd152)
- Nuke the highlighted values
- Now run the normal uninstaller and reinstall it with the setup, this should re-create the default states
- ???
- Profit!

**Comment by alphaniner** [comment]
Thanks! Can't do it right now, but when I do I'll report the outcome.

**Comment by nefarius** [comment]
Any luck @alphaniner ?

**Comment by alphaniner** [comment]
Crap, has it been two weeks?!? Sorry.

Anyway I just got around to trying it. No joy, issue persists. But I double-checked Device Manager and it turns out I was wrong, `\Device\HardDiskVolume10` is the correct 'path' for the partition. I also did some more testing, and the issue occurs with **all** the partitions/volumes on two of my disks, even if I had never added an .exe to HidHide before.

Both of these drives are "Dynamic Disks" while the others (three of them) are "Basic Disks". This is because I have a Windows RAID0 (striped volume) on a partition on each of these two. There is also one additional partition on the first and three additional partitions on the second.
![Capture](https://github.com/nefarius/HidHide/assets/8930509/449cb94e-65d9-4dca-b1de-9fa73309b9e9)

All this being said I had been planning to reinstall Windows soon™ anyway. I'll probably keep the old install around for a while, so if you think of anything let me know. Or feel free to close if you don't think a solution is likely. Either way, thanks for the help.

**Comment by nefarius** [comment]
No worries, 2 weeks is nothing in my time line 😅

I never tested the behaviour with dynamic disks, there has to be a difference in how the kernel resolves the drive names. I'll see if I can replicate it.

**Comment by nefarius** [comment]
Yeah guess what; haven't had the time to look into this, if someone wants to jump on the wagon and help out here, would be cool.
